kratos/linear_solvers/fallback_linear_solver.h

```diff
@@ -0,0 +1,895 @@
+     * @param ThisParameters The configuration parameters
+     */
+    FallbackLinearSolver(
+        LinearSolverPointer pSolver1,
+        LinearSolverPointer pSolver2,
+        Parameters ThisParameters = Parameters(R"({})")
+        ) : mSolvers({pSolver1, pSolver2}),
+            mParameters(ThisParameters)
+    {
+        // Verify that linear solvers are not defined in the parameters
+        KRATOS_ERROR_IF(mParameters.Has("solvers")) << "The solvers are already defined in the input parameters" << std::endl;
+
+        // Set the default parameters
+        mParameters.ValidateAndAssignDefaults(GetDefaultParameters());
+
+        // Fill the parameters with the solvers
+        FillParametersFromSolver(pSolver1);
+        FillParametersFromSolver(pSolver2);
+
+        // Set some member variables from the parameters
+        CommonSettingsFromParameters();
+    }
+
+    /**
+     * @brief Constructor with a list of solvers
+     * @param rSolvers A vector of LinearSolverPointer to set.
+     * @param ThisParameters The configuration parameters
+     */
+    FallbackLinearSolver(
+        const std::vector<LinearSolverPointer>& rSolvers,
+        Parameters ThisParameters = Parameters(R"({})")
+        ) : mSolvers(rSolvers),
+            mParameters(ThisParameters)
+    {
+        // Verify that linear solvers are not defined in the parameters
+        KRATOS_ERROR_IF(mParameters.Has("solvers")) << "The solvers are already defined in the input parameters" << std::endl;
+
+        // Set the default parameters
+        mParameters.ValidateAndAssignDefaults(GetDefaultParameters());
+
+        // Fill the parameters with the solvers
+        for (auto& p_solver : mSolvers) {
+            FillParametersFromSolver(p_solver);
+        }
+
+        // Set some member variables from the parameters
+        CommonSettingsFromParameters();
+    }
+
+    /// Copy constructor.
+    FallbackLinearSolver(const FallbackLinearSolver& rOther) = delete;
+
+    /// Destructor.
+    ~FallbackLinearSolver() override = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /// Assignment operator.
+    FallbackLinearSolver& operator=(const FallbackLinearSolver& rOther) = delete;
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * @brief This function is designed to be called as few times as possible.
+     * @details It creates the data structures that only depend on the connectivity of the matrix (and not on its coefficients) so that the memory can be allocated once and expensive operations can be done only when strictly  needed
+     * @param rA. System matrix
+     * @param rX. Solution vector. it's also the initial guess for iterative linear solvers.
+     * @param rB. Right hand side vector.
+     */
+    void Initialize(
+        SparseMatrixType& rA,
+        VectorType& rX,
+        VectorType& rB
+        ) override
+    {
+        // TODO: Decide if call all solvers or just the current one
```

Comment:
how about making this an input for the FallBacksolver?

Reply:
Could be, but I would default it as it is

Reply:
I am thinkig that maybe the most consistent thing would be to call these methods again once the solver fails

Reply:
We assume that initializes internal data of solvers, not system of equations. Therefore should eb called in all solvers.